doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1257,409 +1261,344 @@ Managing the local context
-      This is equivalent to :n:`pose (@ident := fun {* @binder } => @term)`.
-
-   .. tacv:: pose @term
-
-      This behaves as :n:`pose (@ident := @term)` but :token:`ident` is
-      generated by Coq.
-
-   .. tacv:: epose (@ident {* @binder } := @term)
-             epose @term
+   .. tacn:: epose @alias_definition
+             epose @one_term {? @as_name }
       :name: epose; _
 
-      While the different variants of :tacn:`pose` expect that no
-      existential variables are generated by the tactic, :tacn:`epose`
-      removes this constraint.
-
-.. tacn:: decompose [{+ @qualid}] @term
-   :name: decompose
-
-   This tactic recursively decomposes a complex proposition in order to
-   obtain atomic ones.
-
-   .. example::
-
-      .. coqtop:: reset all
-
-         Goal forall A B C:Prop, A /\ B /\ C \/ B /\ C \/ C /\ A -> C.
-           intros A B C H; decompose [and or] H.
-           all: assumption.
-         Qed.
-
-   .. note::
-
-      :tacn:`decompose` does not work on right-hand sides of implications or
-      products.
+      Similar to :tacn:`pose`, but instead of failing because of uninstantiated
+      variables, generates existential variables for them.
 
-   .. tacv:: decompose sum @term
-
-      This decomposes sum types (like :g:`or`).
-
-   .. tacv:: decompose record @term
-
-      This decomposes record types (inductive types with one constructor,
-      like :g:`and` and :g:`exists` and those defined with the :cmd:`Record`
-      command.
+.. todo: the following title seems inappropriate.  How about something
+   more like "Introducing new hypotheses", as in adding arbitrary terms rather
+   than transformations of existing terms??  But then I think the tactics in the
+   previous section (set, remember, pose, maybe decompose) should be moved into
+   this section.  But maybe hard to make the section seem like an crisp, intuitive grouping.
+   I can do the moving that after we've reviewed all the text.  WDYT?
 
+   See https://github.com/coq/coq/pull/16498#discussion_r989928078
 
 .. _controllingtheproofflow:
 
 Controlling the proof flow
 ------------------------------
 
-.. tacn:: assert (@ident : @type)
-   :name: assert
-
-   This tactic applies to any goal. :n:`assert (H : U)` adds a new hypothesis
-   of name :n:`H` asserting :g:`U` to the current goal and opens a new subgoal
-   :g:`U` [2]_. The subgoal :g:`U` comes first in the list of subgoals remaining to
-   prove.
-
-   .. exn:: Not a proposition or a type.
-
-      Arises when the argument :token:`type` is neither of type :g:`Prop`,
-      :g:`Set` nor :g:`Type`.
+.. tacn:: assert ( @ident : @type ) {? by @ltac_expr3 }
+          assert ( @ident := @term )
+          assert @one_term {? @as_ipat } {? by @ltac_expr3 }
+   :name: assert; _; _
 
-   .. tacv:: assert @type
+   Adds a new hypothesis to the current subgoal and a
+   new subgoal to prove the hypothesis.  The new subgoal becomes
+   the first in the list of subgoals.  Then, if :n:`@ltac_expr3`
```

Comment:
I realize only now that there might be a misunderstanding with this "becomes the first in the list of subgoals". If it happens that there are several goals in focus and `assert` is applied to the second one (with `2:`), then, the new subgoal is inserted just before the subgoal applied was to, and does not become the first in the list of subgoals. Maybe to avoid this misunderstanding, it would be better to say, adapting what you did for `enough`:
```suggestion
   Adds a new hypothesis to the current subgoal and a new subgoal before
   it to prove the hypothesis.  Then, if :n:`@ltac_expr3`
```

Reply:
Got it.

IIUC, this seems like something that could be mostly described in proof context section, e.g. the subgoals resulting from applying a tactic to a goal will appear together in the context where the goal was.  If the tactic adds additional subgoals, they are usually added after the modified goal.  OTOH, this behavior is obvious and almost subliminal, but very noticeable if it's not done that way.